src/pybind/mgr/dashboard_v2/controllers/rgw.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import json
+
+from ..tools import ApiController, RESTController
+from .. import logger
+
+
+@ApiController('rgw')
+class Rgw(RESTController):
+
+    def __init__(self):
+        self.daemon = RgwDaemon()
+
+
+@ApiController()
+class RgwDaemon(RESTController):
+
+    def list(self):
+        daemons = []
+        for server in self.mgr.list_servers():
+            for service in server['services']:
+                if service['type'] == 'rgw':
+                    metadata = self.mgr.get_metadata('rgw', service['id'])
+                    status = self.mgr.get_daemon_status('rgw', service['id'])
+                    if 'json' in status:
+                        try:
+                            status = json.loads(status['json'])
+                        except ValueError:
+                            msg = "{0} had invalid status json"
+                            logger.warning(msg.format(service['id']))
+                            status = {}
+                    else:
+                        msg = '{0} has no key "json" in status'
+                        logger.warning(msg.format(service['id']))
+
+                    # extract per-daemon service data and health
+                    daemon = {
+                        'id': service['id'],
+                        'version': metadata['ceph_version'],
+                        'server_hostname': server['hostname'],
+                        'service': service,
+                        'server': server,
+                        'metadata': metadata,
+                        'status': status,
+                        'url': "{0}/api/rgw/daemon/{1}".format(
+                            self.mgr.url_prefix, service['id'])
+                    }
+
+                    daemons.append(daemon)
+
+        return {'daemons': sorted(daemons, key=lambda k: k['id'])}
+
+    def get(self, service_id):
```

Comment:
I think this can be reduced to the following. The sorting at the end is taken from the origin code.
```
    def get(self, svc_id):
        rgw_metadata = {}
        rgw_status = {}

        for daemon in self.list():
            if daemon["id"] != svc_id:
                continue

            rgw_metadata = daemon["metadata"]
            rgw_status = daemon["status"]

        return {
            "rgw_id": svc_id,
            "rgw_metadata": sorted(rgw_metadata.iteritems()),
            "rgw_status": sorted(rgw_status.iteritems())
        }
```

Reply:
No, that won't work, because the key `status` doesn't seem to be there all the time. In fact it wasn't there once since I started this implementation but since the current implementation is basically the port of the `dashboard` module, I kept it purposely so that it works if - in some case, the future or for a reason I don't know yet - the status will be returned.